docs/docs/multirole.md

```diff
@@ -0,0 +1,337 @@
+
+{
+  "specRole": "dev",
+  "instanceRoles": [
+    "slave_public"
+  ]
+}
+```
+
+We can see our spec's role was updated, but the instance we launched prior to the update continues to use the role "slave_public". This means our reservation was preserved.
+
+Let's scale up the pod!
+
+```
+cat <<EOF | curl -v -X PUT http://localhost:8080/v2/pods/dev/bigbusiness?force=true -H "Content-Type: application/json" --data "@-"
+{
+  "id": "/dev/bigbusiness",
+  "scaling": {"instances": 2, "kind": "fixed"},
+  "containers": [
+    {
+      "name": "sleep1",
+      "exec": { "command": { "shell": "pwd; ls -la; date >> le-data/launches.txt; sleep 1000" } },
+      "resources": { "cpus": 0.1, "mem": 32 },
+      "volumeMounts": [{
+        "name": "data",
+        "mountPath": "le-data"
+      }]
+    }
+  ],
+  "volumes": [
+    {
+      "name": "data",
+      "persistent": {"size": 16, "type": "root"}
+    }
+  ],
+  "role": "dev"
+}
+EOF
+```
+
+Then, let's check the role of our instances now that we have a freshly added instance after the role change:
+
+```
+curl http://localhost:8080/v2/pods/dev/bigbusiness::status | jq '{role: .spec.role, instanceRoles: [.instances[].role]}'
+
+{
+  "role": "dev",
+  "instanceRoles": [
+    "slave_public",
+    "dev"
+  ]
+}
+```
+
+We can see that the new pod instance has the new role, while the old pod instance continues to have the old role. Moving forward, all new instances will receive the new role, "dev".
+
+# Group role enforcement
+
+Marathon allows the mixed role usage in order to support a reasonable, incremental upgrade pathway for users wishing to migrate their existing Marathon instances to use multi-role. However, as stated in the introduction, this isn't the ideal. Ideally, Marathon is configured to enforce services to use the group role. This is why the `enforceRole` property exists on groups.
+
+## Using enforceRole
+
+The group `enforceRole` property is settable only for top-level groups; when enabled, it forces all **new** services in that group (or in subgroups) to use the group role, and receive the group role as a default. Existing services still must be migrated to multi-role, by modifying the respective "role" field for each of the existing services.
+
+As an example, let's say we have deployed the following bigbusiness app, using the default role:
+
+```
+{
+  "id": "/dev/bigbusiness",
+  "cmd": "set; sleep 3600",
+  "instances": 10,
+  "cpus": 0.05,
+  "mem": 128,
+  "role": "slave_public"
+}
+```
+
+Now, we use the PATCH method to enable role enforcement for a group:
+
+```
+$ echo '{"enforceRole": true}' | curl -f -X PATCH http://localhost:8080/v2/groups/dev -H "Content-Type: application/json" --data "@-"
+```
+
+Note: the API will simply respond with an empty entity, with a 200 status if the operation is successful. We can check the value for the `enforceRole` field:
+
+```
+
+$ curl http://localhost:8080/v2/groups/dev | jq .enforceRole
+
+true
+```
```

Comment:
Why are there blank lines before and after the curl?

Reply:
It's the output